core_js/popup.js

```diff
@@ -62,6 +62,40 @@ function changeStatistics()
     elTotal.textContent = totalCounter.toLocaleString();
 }
 
+/**
+* Set the whitelist button text
+*/
+function setWhitelistText()
+{
+    let element = document.getElementById('whitelist_btn');
+    let currentSite, siteFound;
+    browser.tabs.query({active: true, currentWindow: true}, function(tabs) {
+        currentSite = tabs[0].url;
+    });
+
+    browser.runtime.sendMessage({
+        function: "getData",
+        params: ['whitelist']
+    }).then((data) => {
+        data.response.forEach(site => {
```

Comment:
Should be replaced by the following: `let siteFound = data.response.some(site => currentSite.indexOf(site) !== -1);`